wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -438,21 +437,20 @@ private FactoryDto asDto(Factory factory) throws ServerException {
     @Inject private DefaultFactoryParameterResolver defaultFactoryResolver;
 
     /**
-     * Provides a suitable resolver for the given parameters
+     * Provides a suitable resolver for the given parameters or {@code null} if none is found
+     * (generally means that provided parameters set is invalid)
```

Comment:
Should we throw an Exception if provided parameters are invalid?

Reply:
it is already thrown in caller method. but yes, it looks better to throw it here instead of just returning `null`